bundles/org.smarthomej.binding.androiddebugbridge/src/main/java/org/smarthomej/binding/androiddebugbridge/internal/AndroidDebugBridgeDevice.java

```diff
@@ -258,42 +257,19 @@ public boolean isAwake() throws InterruptedException, AndroidDebugBridgeDeviceEx
 
     public boolean isScreenOn() throws InterruptedException, AndroidDebugBridgeDeviceException,
             AndroidDebugBridgeDeviceReadException, TimeoutException, ExecutionException {
-        String result = runAdbShell("dumpsys", "power", "|", "grep", "'Display Power'");
-        String[] splitResult = result.split("=");
-        if (splitResult.length >= 2) {
-            return "ON".equals(splitResult[1]);
+        String result = runAdbShell("dumpsys", "display", "|", "grep", "mScreenState");
+        if (result.contains("mScreenState=")) {
+            return result.contains("mScreenState=ON");
         }
         throw new AndroidDebugBridgeDeviceReadException(SCREEN_STATE_CHANNEL, result);
     }
 
-    public Optional<Boolean> isHDMIOn() throws InterruptedException, AndroidDebugBridgeDeviceException,
+    public boolean isHdmiOn() throws InterruptedException, AndroidDebugBridgeDeviceException,
             AndroidDebugBridgeDeviceReadException, TimeoutException, ExecutionException {
-        if (channelFallbackMap.get(HDMI_STATE_CHANNEL) == FallbackModes.LOGCAT) {
-            return isHDMIOnWithLogcat();
-        }
-        String result = runAdbShell("cat", "/sys/devices/virtual/switch/hdmi/state");
-        if ("0".equals(result) || "1".equals(result)) {
-            return Optional.of("1".equals(result));
-        } else {
-            LOGGER.debug("set fallback {} for {}", FallbackModes.LOGCAT, HDMI_STATE_CHANNEL);
-            channelFallbackMap.put(HDMI_STATE_CHANNEL, FallbackModes.LOGCAT);
-            return isHDMIOnWithLogcat();
+        String result = runAdbShell("dumpsys", "hdmi_control", "|", "grep", "mPowerStatus");
```

Comment:
So this works for both cases we had before?

Reply:
Yes, it is much more stable than the previous way(s).